zoneinfo/tests/test_datagen.rs

```diff
@@ -0,0 +1,218 @@
+#[cfg(feature = "std")]
+use std::{
+    format,
+    fs::{self, read_to_string},
+    path::{Path, PathBuf},
+    string::String,
+    vec::Vec,
+};
+#[cfg(feature = "std")]
+use zoneinfo_rs::{ZoneInfoCompiler, ZoneInfoData};
+
+#[cfg(feature = "std")]
+#[derive(Debug, Serialize, Deserialize)]
+struct TzifTestData {
+    first_record: LocalRecord,
+    transitions: Vec<TransitionRecord>,
+}
+
+#[cfg(feature = "std")]
+#[derive(Debug, Serialize, Deserialize)]
+struct TransitionRecord {
+    transition_time: i64,
+    record: LocalRecord,
+}
+
+#[cfg(feature = "std")]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct LocalRecord {
+    offset: i64,
+    is_dst: bool,
+    abbr: String,
+}
+
+// Utility function for generating example files
+#[allow(unused)]
+#[cfg(feature = "std")]
+fn generate_test_data(tzdata_dir: PathBuf, identifier: &str) {
+    let manifest_dir = Path::new(env!("CARGO_MANIFEST_DIR"));
+    let test_dir = manifest_dir.join("tests");
+    let filename = identifier.to_lowercase().replace("/", "-");
+    let test_data_path = test_dir.join(format!("{filename}.json"));
+
+    let tzdata = manifest_dir.join("tzdata/test/");
+    let tzif = tzif::parse_tzif_file(&tzdata_dir.join(identifier)).unwrap();
+    let tzif_block_v2 = tzif.data_block2.unwrap();
+
+    let first_record_data = tzif_block_v2.local_time_type_records[0];
+    let first_record = LocalRecord {
+        offset: first_record_data.utoff.0,
+        is_dst: first_record_data.is_dst,
+        abbr: tzif_block_v2.time_zone_designations[0].clone(),
+    };
+
+    print!("{:#?}", tzif_block_v2);
+
+    let local_records = tzif_block_v2
+        .local_time_type_records
+        .iter()
+        .enumerate()
+        .map(|(idx, r)| LocalRecord {
+            offset: r.utoff.0,
+            is_dst: r.is_dst,
+            abbr: tzif_block_v2
+                .time_zone_designations
+                .get(r.idx / 4)
+                .cloned()
+                .unwrap_or(String::from("unknown")),
+        })
+        .collect::<Vec<_>>();
+
+    let transitions = tzif_block_v2
+        .transition_times
+        .iter()
+        .zip(tzif_block_v2.transition_types)
+        .map(|(time, time_type)| TransitionRecord {
+            transition_time: time.0,
+            record: local_records[time_type].clone(),
+        })
+        .collect::<Vec<TransitionRecord>>();
+
+    let tzif_data = TzifTestData {
+        first_record,
+        transitions,
+    };
+
+    std::println!("Writing generated example data to {:?}", test_data_path);
+    fs::write(
+        test_data_path,
+        serde_json::to_string_pretty(&tzif_data).unwrap(),
+    )
+    .unwrap();
+}
+
+// Uncomment and adjust path to generate any new testing data
+//
+// #[test]
+// #[cfg(feature = "std")]
+// fn gen() {
+//     let manifest_dir = Path::new(env!("CARGO_MANIFEST_DIR"));
+//     let path = manifest_dir.join("tzdata/test/");
+//     generate_test_data(path, "Antarctica/Troll");
+// }
```

Comment:
This should probably be an external tool like `diplomat-gen`, more so because we could automatically check that the test data is synced with the tzif data.

Reply:
Not against making it an external tool, but it is currently a bit flawed. Probably a better way to manage it.

> we could automatically check that the test data is synced with the tzif data.

What tzif data is being referred to here as far as syncing?

Reply:
The tzif data that is used to generate the JSON test data

Reply:
Yeah, sourcing that tzif data is going to be the big question there though :laughing:

Reply:
Is the `zic` binary tied to the tzdb version? Because we could require that you install that binary, then the script would download a pinned version of tzdb to compile it to a tzif

Reply:
`zic` is not pinned to the version, but it has various compilation flags. For instance, currently the files we have in the repo are bloat=SLIM and vanguard. I'm on an Ubuntu distro and that's bloat=FAT and rearguard.

We could download the zoneinfo files, but then that probably should be behind a "networking" feature at some point, which is why the data is precompiled here.

Reply:
> We could download the zoneinfo files, but then that probably should be behind a networking feature at some point, which is why the data is precompiled here in the repo.

I don't know if we need a flag if the tool will only be used internally to generate test data... Though, if we decide to expose the tool, I agree that it would be better to have a `networking` feature.

Reply:
Yeah, I understood what you meant. I think the networking thing is just general hesitation on my part leaking in :laughing: 

I should be able to pull it out later tonight. It would be nicer to work with as some tooling vs. running test (I've just been enjoying clicking the VS Code "Run test" button).

Reply:
Moved it out and added a little bit to a README file for the tool.